lib/mix/test/mix/cli_test.exs

```diff
@@ -198,6 +198,67 @@ defmodule Mix.CLITest do
     System.delete_env("MIX_EXS")
   end
 
+  test "env target", %{tmp_dir: tmp_dir} do
+    File.cd!(tmp_dir, fn ->
+      File.write!("custom.exs", """
+      defmodule P do
+        use Mix.Project
+        def project, do: [app: :p, version: "0.1.0"]
+      end
+      """)
+
+      System.put_env("MIX_TARGET", "rpi0")
+      System.put_env("MIX_EXS", "custom.exs")
+
+      output = mix(["run", "-e", "IO.inspect {Mix.target(), System.argv()}", "--", "a", "b", "c"])
+
+      assert output =~ ~s({:rpi0, ["a", "b", "c"]})
+    end)
+  after
+    System.delete_env("MIX_TARGET")
+    System.delete_env("MIX_EXS")
+  end
+
+  test "env target defaults to host", %{tmp_dir: tmp_dir} do
+    File.cd!(tmp_dir, fn ->
+      File.write!("custom.exs", """
+      defmodule P do
+        use Mix.Project
+        def project, do: [app: :p, version: "0.1.0"]
+      end
+      """)
+
+      System.put_env("MIX_EXS", "custom.exs")
+
+      output = mix(["run", "-e", "IO.inspect {Mix.target(), System.argv()}", "--", "1", "2", "3"])
+
+      assert output =~ ~s({:host, ["1", "2", "3"]})
+    end)
+  after
+    System.delete_env("MIX_EXS")
+  end
+
+  test "env target defaults to host even when empty", %{tmp_dir: tmp_dir} do
+    File.cd!(tmp_dir, fn ->
+      File.write!("custom.exs", """
+      defmodule P do
+        use Mix.Project
+        def project, do: [app: :p, version: "0.1.0"]
+      end
+      """)
+
+      System.put_env("MIX_TARGET", "")
+      System.put_env("MIX_EXS", "custom.exs")
+
+      output = mix(["run", "-e", "IO.inspect {Mix.target(), System.argv()}", "--", "a", "b", "c"])
+
+      assert output =~ ~s({:host, ["a", "b", "c"]})
+    end)
+  after
+    System.delete_env("MIX_TARGET")
+    System.delete_env("MIX_EXS")
+  end
```

Comment:
Those tests are expensive so we tend to have few of them. Can you please keep only this last test and remove the other two? While at it, please check for `System.put_env("MIX_ENV", "")` too. Thank you!

Reply:
Done.